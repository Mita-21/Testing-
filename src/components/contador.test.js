import { render, fireEvent} from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import Botonmenos from './botonmenos';
import Botonmás from './botonmás';
import Contador from './contador';

describe('my tests', () => {

test('should button to have plus icon', () => {

  const component = render(
    <Contador />
  )
  const button = component.getByText('+')

  expect(button).toBeTruthy()

  }); 
  test('should button to have minus icon', () => {

    const component = render(
      <Contador />
    )
    const button = component.getByText('-')
  
    expect(button).toBeTruthy()
  
    }); 

  test('increments count by one when + is clicked', () => {
    
  const component = render(
    <Contador/>
  )

  fireEvent.click(component.getByText('+'))
  fireEvent.click(component.getByText('+'))

  let num = component.getByText('2')
  expect(num).toBeInTheDocument()
  
  })
  test('decrements count by one when - is clicked', () => {
    
    const component = render(
      <Contador/>
    )
  
    fireEvent.click(component.getByText('-'))
   
  
    let num = component.getByText('-1')
    expect(num).toBeInTheDocument()
    
    })
    test('clicking increment button calls event handler once', () => {
      const mockHandler = jest.fn()

      const component = render(
        <Botonmás increment={mockHandler} />
      )

      const button = component.getByText('Incrementar')
    fireEvent.click(button)

  expect(mockHandler.mock.calls).toHaveLength(1)

    })
    test('clicking decrement button calls event handler once', () => {
      const mockHandler = jest.fn()

      const component = render(
        <Botonmenos decrement={mockHandler} />
      )

      const button = component.getByText('Decrementar')
    fireEvent.click(button)

  expect(mockHandler.mock.calls).toHaveLength(1)

    })
    test('practice snapshot', () => {
      
      const component = renderer.create(
     <Contador></Contador>
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
  
    });

});