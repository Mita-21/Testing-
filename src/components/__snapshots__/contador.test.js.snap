// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my tests practice snapshot 1`] = `
<div>
  <h1>
    Hello React World
  </h1>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Contador
      </div>
      <div
        className="meta"
      >
        0
      </div>
      <div
        className="description"
      >
        Presiona - รณ + para modificar
      </div>
    </div>
    <div
      className="extra content"
    >
      <div
        className="ui two buttons"
      >
        <button
          className="ui green basic button"
          onClick={[Function]}
        >
          -
        </button>
        <button
          className="ui red basic button"
          onClick={[Function]}
        >
          +
        </button>
      </div>
    </div>
    <div
      className="content"
    >
      <div>
        <div>
          <button
            onClick={[Function]}
          >
            Incrementar
          </button>
        </div>
      </div>
      <div>
        <div>
          <button
            onClick={[Function]}
          >
            Decrementar
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
